src/ObjectUtils/ObjectFileTest.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "ObjectUtils/ObjectFile.h"
+#include "OrbitBase/ExecutablePath.h"
+
+using ::orbit_object_utils::CreateObjectFile;
+
+// Only tests methods that are in the interface for ObjectFile itself. More detailed tests
+// specific to ElfFile and CoffFile are in their own tests.
+
+TEST(ObjectFile, CorrectObjectTypeForElf) {
+  std::filesystem::path file_path =
+      orbit_base::GetExecutableDir() / "testdata" / "hello_world_elf_with_debug_info";
+
+  auto object_file = CreateObjectFile(file_path);
+  ASSERT_TRUE(object_file.has_value()) << object_file.error().message();
```

Comment:
You can now use ASSERT_THAT(object_file, HasNoError()) which prints error message automatically:
https://github.com/google/orbit/blob/a1a9968cd38977dbb1c38c64a31a0989255dbf0e/src/OrbitBase/include/OrbitBase/TestUtils.h

Reply:
Thanks, that's useful. Replaced here and in other tests in ObjectUtils.